File: app/src/dev/java/com/mycompany/myapp/ui/devsettings/DevSettingsFragment.java
package com.mycompany.myapp.ui.devsettings;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.mycompany.myapp.R;

import javax.inject.Inject;

public class DevSettingsFragment extends Fragment {
    public interface DevSettingsFragmentHost {
        void inject(DevSettingsFragment fragment);
    }

    @Inject DevSettingsPresenter presenter;
    private DevSettingsFragmentBinding binding;
    private DevSettingsFragmentHost host;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        host = (DevSettingsFragmentHost) context;
        host.inject(this);
    }

    @Override
    public void onDetach() {
        host = null;
        super.onDetach();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_dev_settings, container, false);
        binding.setPresenter(presenter);

        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        binding.unbind();
        super.onDestroyView();
    }
}

File: app/src/dev/java/com/mycompany/myapp/ui/devsettings/DevSettingsFragment.kt
package com.mycompany.myapp.ui.devsettings

import android.content.Context
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.mycompany.myapp.R
import com.mycompany.myapp.ui.BaseFragment

class DevSettingsFragment : BaseFragment() {
    interface DevSettingsFragmentHost

    private lateinit var viewModel: DevSettingsViewModel
    private lateinit var binding: DevSettingsFragmentBinding
    private var host: DevSettingsFragmentHost? = null

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        host = context as DevSettingsFragmentHost
    }

    override fun onDetach() {
        host = null
        super.onDetach()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        viewModel = getViewModel(DevSettingsViewModel::class)

        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_dev_settings, container, false)
        binding.vm = viewModel

        return binding.root
    }

    override fun onDestroyView() {
        binding.unbind()
        super.onDestroyView()
    }
}
